module/batch-action/src/Application/Controller/Api/EndBatchAction.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\BatchAction\Application\Controller\Api;
+
+use Symfony\Component\Routing\Annotation\Route;
+use Ergonode\SharedKernel\Domain\Bus\CommandBusInterface;
+use Swagger\Annotations as SWG;
+use Ergonode\BatchAction\Domain\Entity\BatchActionId;
+use Ergonode\BatchAction\Domain\Entity\BatchAction;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Ergonode\BatchAction\Domain\Command\EndBatchActionCommand;
+
+/**
+ * @Route(
+ *     name="ergonode_batch_action_end",
+ *     path="/batch-action/{action}/end",
+ *     methods={"PUT"},
+ *     requirements={"action" = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"}
+ * )
+ */
+class EndBatchAction
+{
+    private CommandBusInterface $commandBus;
+
+    public function __construct(CommandBusInterface $commandBus)
+    {
+        $this->commandBus = $commandBus;
+    }
+
+    /**
+     * @SWG\Tag(name="Batch action")
+     * @SWG\Parameter(
+     *     name="language",
+     *     in="path",
+     *     type="string",
+     *     description="Language code",
+     *     default="en_GB"
+     * )
+     * @SWG\Parameter(
+     *     name="action",
+     *     in="path",
+     *     type="string",
+     *     description="Batch action id",
+     * )
+     * @SWG\Response(
+     *     response=200,
+     *     description="Returns batch information",
+     * )
+     *
+     * @ParamConverter(class="Ergonode\BatchAction\Domain\Entity\BatchAction")
+     */
+    public function __invoke(BatchAction $action): BatchActionId
+    {
+        $this->commandBus->dispatch(new EndBatchActionCommand($action->getId()));
```

Comment:
I think appropriate status should be validated. Otherwise we'd say "ok" for i.e. precessed action which is left in such status.